### ISSUE TITLE
Scrolling commands menu eats some fps

### ISSUE THREAD

**Issue by TheHackerLxr** [kohls-admin/kohls-admin#47]
https://github.com/user-attachments/assets/f2738dc4-8cf3-4996-aa7f-0323d40a0633

When I scroll through the command menu, it eats up my fps, the fps drops from 240 to 90, but it depends on how hard I scroll

**Comment by RyanLua** [comment]
I've been able to reproduce this issue. From the MicroProfiler, it looks like `Render/PreRender/UpdateUILayouts` is taking nearly all of the rendering time.

The documentation proposes the solution:
> Reduce the amount of UI elements being resized or repositioned dynamically, such as those managed by UILayout.
> https://create.roblox.com/docs/studio/microprofiler/tag-table#ui

This problem seems to be with the UI overall and doesn't have an easy fix. Whenever the UI is updated/redrawn, it's really costly to compute that because of its heavy usage of [size modifiers and constraints](https://create.roblox.com/docs/ui/size-modifiers),

![image](https://github.com/user-attachments/assets/c8aa97fc-02f7-4e05-8472-24bf7fea1831)
![image](https://github.com/user-attachments/assets/58096a8b-3ce2-4396-a72a-13f1c070a832)

Taking a deeper dive, it looks like the dashboard renders hundreds of `UIListLayout`s and `GuiObject`s that otherwise could be reduced or removed.

![image](https://github.com/user-attachments/assets/b9984b11-2293-4dd0-8ef2-d98d59d8f838)

TLDR; this is a fundamental problem with how the UI is being rendered and is made.

**Comment by Lucatacos** [comment]
I don’t understand why the reduction of UIListLayouts through static UIs or other performance-enhancing solutions hasn't been addressed yet. Improving performance should be a priority in order to enhance the player experience and overall development. This issue seems fundamental to the UI rendering process and needs attention.

**Comment by TheHackerLxr** [comment]
> I don’t understand why the reduction of UIListLayouts through static UIs or other performance-enhancing solutions hasn't been addressed yet. Improving performance should be a priority in order to enhance the player experience and overall development. This issue seems fundamental to the UI rendering process and needs attention.

It was for old version, now it's just outdated thing. But if you encounter lags or issues with UI scrolling, you can make a new issue about this problem

**Comment by Lucatacos** [comment]
> > I don’t understand why the reduction of UIListLayouts through static UIs or other performance-enhancing solutions hasn't been addressed yet. Improving performance should be a priority in order to enhance the player experience and overall development. This issue seems fundamental to the UI rendering process and needs attention.
> 
> It was for old version, now it's just outdated thing. But if you encounter lags or issues with UI scrolling, you can make a new issue about this problem

Why create a new issue if this was already reported, even if it's outdated?
The problem hasn't changed , it's still the same one you reported before.

![Image](https://github.com/user-attachments/assets/fe4d961d-e5f2-4582-bfec-5e392bedcbfd)

**Comment by TheHackerLxr** [comment]
> > > I don’t understand why the reduction of UIListLayouts through static UIs or other performance-enhancing solutions hasn't been addressed yet. Improving performance should be a priority in order to enhance the player experience and overall development. This issue seems fundamental to the UI rendering process and needs attention.
> > 
> > It was for old version, now it's just outdated thing. But if you encounter lags or issues with UI scrolling, you can make a new issue about this problem
> 
> Why create a new issue if this was already reported, even if it's outdated?
> The problem hasn't changed , it's still the same one you reported before.
> 
> ![Image](https://github.com/user-attachments/assets/fe4d961d-e5f2-4582-bfec-5e392bedcbfd)

If it's not fixed, then we need to wait until other contributors would review this problem. In latest version I dont have such fps drops.

**Comment by kohltastrophe** [comment]
I plan to rehaul the UI in the future when I refactor everything to use [vide](https://centau.github.io/vide/). I'll be dynamically creating components whenever the tab/window is focused/opened to reduce the performance cost.
